ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/electra/statetransition/epoch/EpochProcessorElectra.java

```diff
@@ -322,12 +322,18 @@ public void processPendingConsolidations(final MutableBeaconState state) {
         break;
       }
 
-      final UInt64 activeBalance =
-          stateAccessorsElectra.getActiveBalance(state, pendingConsolidation.getSourceIndex());
+      // Calculate the consolidated balance
+      final UInt64 sourceEffectiveBalance =
+          state
+              .getBalances()
+              .get(pendingConsolidation.getSourceIndex())
+              .get()
+              .min(sourceValidator.getEffectiveBalance());
+      // Move active balance to target. Excess balance is withdrawable.
       beaconStateMutators.decreaseBalance(
-          state, pendingConsolidation.getSourceIndex(), activeBalance);
+          state, pendingConsolidation.getSourceIndex(), sourceEffectiveBalance);
       beaconStateMutators.increaseBalance(
-          state, pendingConsolidation.getTargetIndex(), activeBalance);
+          state, pendingConsolidation.getTargetIndex(), sourceEffectiveBalance);
```

Comment:
Just to make it easier for others to review. These are the spec changes:

<img width="796" alt="image" src="https://github.com/user-attachments/assets/96b8158b-91f3-4e1e-9def-ac2a41454ecc" />

And these changes track those properly 👍

Reply:
yep next time i think this can be a PR and refactor can be separate for sanity.

Reply:
Yes, noted for next time.